pvlib/test/test_atmosphere.py

```diff
@@ -128,6 +128,21 @@ def test_first_solar_spectral_correction_ambiguous():
         atmosphere.first_solar_spectral_correction(1, 1)
 
 
+def test_first_solar_spectral_correction_range():
+    with pytest.warns(UserWarning):
+        out = atmosphere.first_solar_spectral_correction(np.array([.1, 3, 10]),
+                                                         np.array([1, 3, 5]),
+                                                         module_type='monosi')
+        expected = np.array([0.96080878, 1.03055092,        nan])
```

Comment:
I believe that this line and the next need to shift left. If `pytest.warns` works like `pytest.raises` the code exits the `with` block as soon as the `UserWarning` is raised, so the `assert_allclose` is never executed.  I could be mistaken.

Reply:
Seems like it is different in this case. I have simply tried by changing the value of 'expected', and the test actually failed. So lines 136 and 137 are executed.

Reply:
Good to know, `warns` behaves different from `raises`

Reply:
While not strictly required, the lines should shift left for style purposes. A `with` block should contain as little code as possible, in this case only the `first_solar_spectral_correction` call.

The difference between the `pytest.warns` and `pytest.raises` requirement is due to what happens within the actual library code (functions don't stop when they emit warnings, they do stop when the raise exceptions) - not anything to do with pytest.